numpy/lib/arraysetops.py

```diff
@@ -519,6 +522,27 @@ def in1d(ar1, ar2, assume_unique=False, invert=False):
     ar1 = np.asarray(ar1).ravel()
     ar2 = np.asarray(ar2).ravel()
 
+    # Make sure we won't encounter memory errors
+    if fast_integers and np.max(ar2) - np.min(ar2) < 8e8:
```

Comment:
I counted four occurences of `np.min(ar2)` and three of `np.max(ar2)`. It would probably make sense not to look up the extrema each time.
Also I keep forgetting: is there any difference between `np.max(ar2)` and `ar2.max()`? Does the former merely call the latter?

Reply:
Good point. And yes if I recall correctly it is just an alias.

Reply:
> is there any difference between np.max(ar2) and ar2.max()? Does the former merely call the latter?

Short answer - they both call the `umath` `ufunc` `numpy.maximum.reduce` via python calls, but via different paths. Long answer:

- The first one calls [`_wrap_reduction(ar2, numpy.maximum, ...)`](https://github.com/numpy/numpy/blob/v1.15.2/numpy/core/fromnumeric.py#L2333) which checks if ar2 is an ndarray, then calls [`numpy.maximum.reduce(ar2, ...)`](https://github.com/numpy/numpy/blob/v1.15.2/numpy/core/fromnumeric.py#L83).

- The second maps to [`array_max`](https://github.com/numpy/numpy/blob/v1.15.2/numpy/core/src/multiarray/methods.c#L317) in C, which ends up calling out to python in [`_methods.py`](https://github.com/numpy/numpy/blob/v1.15.2/numpy/core/_methods.py#L17) and calling `numpy.maximum.reduce(ar2, ...)`.